xdsl/dialects/onnx.py

```diff
@@ -932,6 +932,57 @@ def verify_(self) -> None:
             )
 
 
+@irdl_op_definition
+class Sigmoid(IRDLOperation):
+    """
+    Applies the sigmoid function element-wise to all elements of the input tensor.
+    The sigmoid function, denoted by sigma(x), is a common mathematical function used in machine learning and neural networks. It is defined as:
+    sigma(x) = 1 / (1 + e^-x)
+    where e is the base of the natural logarithm. The sigmoid function maps any real-valued number to the range of [0, 1].
+    The sigmoid function is used as an activation function.
+
+    Args:
+    - input_tensor (TensorType): The input tensor to which the sigmoid function will be applied.
+
+    Returns:
+    - output_tensor (TensorType): The output tensor after applying the sigmoid function element-wise to the input tensor.
+    """
+
+    name = "onnx.Sigmoid"
+
+    T = Annotated[AnyFloat | IntegerType, ConstraintVar("T")]
+    input_tensor = operand_def(TensorType[T])
+    output_tensor = result_def(TensorType[T])
```

Comment:
```suggestion
    T = Annotated[TensorType[AnyFloat], ConstraintVar("T")]
    input_tensor = operand_def(T)
    output_tensor = result_def(T)
```
Is it possible to do sigmoid of integers? It feels safe to only allow floats. The other thing is if the shapes and types are the same, then the types have to be the same, so we can change the declarative constraints, and remove the shape constraint below.

Reply:
Done